src/OrchardCore.Modules/OrchardCore.Search.Elastic/Startup.cs

```diff
@@ -0,0 +1,181 @@
+        private readonly IShellConfiguration _configuration;
+        private readonly ILogger<Startup> _logger;
+
+        public Startup(IOptions<AdminOptions> adminOptions,
+            IShellConfiguration configuration,
+            ILogger<Startup> logger)
+        {
+            _adminOptions = adminOptions.Value;
+            _configuration = configuration;
+            _logger = logger;
+        }
+
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            var url = _configuration[$"OrchardCore_Elastic:{nameof(ElasticConnectionOptions.Url)}"];
+            if (CheckOptions(url, _logger))
+            {
+                var settings = new ConnectionSettings(new Uri(url));
+                var client = new ElasticClient(settings);
+                services.AddSingleton<IElasticClient>(client);
+            }
+            services.AddSingleton<ElasticIndexingState>();
+            services.AddSingleton<ElasticIndexSettingsService>();
+            services.AddSingleton<ElasticIndexManager>();
+            services.AddSingleton<ElasticAnalyzerManager>();
+            services.AddScoped<ElasticIndexingService>();
+            services.AddScoped<ISearchQueryService, SearchQueryService>();
+
+            services.AddScoped<IContentTypePartDefinitionDisplayDriver, ContentTypePartIndexSettingsDisplayDriver>();
+            services.AddScoped<IContentPartFieldDefinitionDisplayDriver, ContentPartFieldIndexSettingsDisplayDriver>();
+            services.AddScoped<INavigationProvider, AdminMenu>();
+            services.AddScoped<IPermissionProvider, Permissions>();
+
+            //services.Configure<LuceneOptions>(o =>
+            //    o.Analyzers.Add(new LuceneAnalyzer(LuceneSettings.StandardAnalyzer,
+            //        new StandardAnalyzer(LuceneSettings.DefaultVersion))));
+
+            services.AddScoped<IDisplayDriver<ISite>, ElasticSettingsDisplayDriver>();
+            services.AddScoped<IDisplayDriver<Query>, ElasticQueryDisplayDriver>();
+
+            services.AddScoped<IContentHandler, ElasticIndexingContentHandler>();
+            services.AddElasticQueries();
+
+            // LuceneQuerySource is registered for both the Queries module and local usage
+            services.AddScoped<IQuerySource, ElasticQuerySource>();
+            services.AddScoped<ElasticQuerySource>();
+            services.AddRecipeExecutionStep<ElasticIndexStep>();
+
+            services.AddScoped<IShapeTableProvider, SearchShapesTableProvider>();
+            services.AddShapeAttributes<SearchShapes>();
+        }
+
+        public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+            routes.MapAreaControllerRoute(
+                name: "Elastic.Search",
+                areaName: "OrchardCore.Search.Elastic",
+                pattern: "Search",
+                defaults: new { controller = "Search", action = "Search" }
```

Comment:
This is conflicting with the Lucene module, does it mean we shouldn't have both enabled at the same time?

Reply:
I think we should move the Search feature out of the Lucene module instead and make it using an abstraction on the Search service.

Reply:
@Skrypt @sebastienros  Perhaps evaluate, if we need to follow something like this

1) OrchardCore.Search (All abstractions related to Search
2) OrchardCore.Indexing (Abstractions for common indexing configurations)
3) OrchardCore.Search.Queries (May need to do away with OrchardCore.Search??)
4) OrchardCore.Search.Components (All Search based components/like drop downs)
5) OrchardCore.Search.Lucene
6) OrchardCore.Search.Elastic
7) OrchardCore.Search.Azure